imap_processing/ultra/l1b/ultra_l1b_annotated.py

```diff
@@ -0,0 +1,50 @@
+"""Calculates Annotated Events for ULTRA L1b."""
+
+import typing
+
+import numpy as np
+import spiceypy as spice
+from numpy.typing import NDArray
+
+from imap_processing.spice.kernels import ensure_spice
+
+
+@ensure_spice
+@typing.no_type_check
+def get_particle_velocity(
+    time: np.ndarray,
+    instrument_velocity: np.ndarray,
+    instrument_frame: str,
```

Comment:
This should require the frame input to be `SpiceFrame` type as defined in spice/geometry.py. It looks like IMAP_DPS is defined in the `SpiceBody` enum but not `SpiceFrame`.